rfcs/20230818-Dropout/README.md

```diff
@@ -0,0 +1,79 @@
+# Introducing Dropout primitive attribute
+
+## Introduction
+
+In many DNN and GNN models, [Dropout](https://en.wikipedia.org/wiki/Convolutional_neural_network#Dropout) is used to improve training results. In some cases, this layer can take a significant amount of time. To enhance the performance of training, we want to optimize it and, as a result, fuse it with the previous primitive.
+
+This idea was [proposed](https://github.com/oneapi-src/oneDNN/pull/760) some time ago. As a result of that discussion, a primitive attribute was proposed to resolve the performance gap.
+
+## Proposal
+
+Additional function to set dropout attibute in C API:
+
+```c
+/// Returns the parameters of a drop out attribute.
+///
+/// @param attr Primitive attributes.
+/// @param enable_drop Drop-out enable/disable flag
+/// @param mask_desc Output memory descriptor of a drop out mask.
+/// @returns #dnnl_success on success and a status describing the error
+///     otherwise.
+dnnl_status_t DNNL_API dnnl_primitive_attr_get_dropout(
+        const_dnnl_primitive_attr_t attr, uint8_t *enable_drop,
+        const_dnnl_memory_desc_t *mask_desc);
+
+/// Set up drop-out primitive attribute.
+///
+/// @param attr Primitive attributes.
+/// @param enable_drop Drop-out enable/disable flag
+/// @param mask_desc Output memory descriptor of a drop out mask.
+/// @returns #dnnl_success on success and a status describing the error
+///     otherwise.
+dnnl_status_t DNNL_API dnnl_primitive_attr_set_dropout(
+        dnnl_primitive_attr_t attr, uint8_t enable_drop,
```

Comment:
I would suggest to remove `enable_drop` from both APIs:
- In getter rely on md returned from the API call. If it is a zero md, then dropout is not set by user.
- In setter -- once user called `dnnl_primitive_attr_set_dropout(attr, mask)` dropout is set.

The current API is a bit confusing when user calls `dnnl_primitive_attr_set_dropout(attr, **false**, mask_desc)`

Reply:
@igorsafo , thanks for you suggestion!
I updated README and PoC